src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -430,112 +543,158 @@ subroutine general_read_fv3_regional(this,fv3_filenameginput,g_ps,g_u,g_v,g_tv,g
-      character(len=:),allocatable :: dynvars   !='fv3_dynvars'
-      character(len=:),allocatable :: tracers   !='fv3_tracer'
-      character(len=:),allocatable :: sfcdata   !='fv3_sfcdata'
-      character(len=:),allocatable :: couplerres!='coupler.res'
-      
-      associate( this => this ) ! eliminates warning for unused dummy argument needed for binding
-      end associate
+    
+    associate( this => this ) ! eliminates warning for unused dummy argument needed for binding
+    end associate
 
 
 
+    call gsi_gridcreate(grid_ens,grd_ens%lat2,grd_ens%lon2,grd_ens%nsig)
     grid_spec=fv3_filenameginput%grid_spec
     ak_bk=fv3_filenameginput%ak_bk
     dynvars=fv3_filenameginput%dynvars
     tracers=fv3_filenameginput%tracers
     sfcdata=fv3_filenameginput%sfcdata
     couplerres=fv3_filenameginput%couplerres
 
-      
 
-!cltthinktobe  should be contained in variable like grd_ens
+     
+     
+    if (allocated(fv3lam_ens_io_dynmetvars2d_nouv) ) then   
+       call gsi_bundlecreate(gsibundle_fv3lam_ens_dynvar_nouv,grid_ens,'gsibundle_fv3lam_ens_dynvar_nouv', istatus,&
+                names2d=fv3lam_ens_io_dynmetvars2d_nouv,names3d=fv3lam_ens_io_dynmetvars3d_nouv)
+    else
+       call gsi_bundlecreate(gsibundle_fv3lam_ens_dynvar_nouv,grid_ens,'gsibundle_fv3lam_ens_dynvar_nouv',istatus, &
+                names3d=fv3lam_ens_io_dynmetvars3d_nouv)
+    endif
+
+     
+    if (allocated(fv3lam_ens_io_tracermetvars2d_nouv) ) then   
+       call gsi_bundlecreate(gsibundle_fv3lam_ens_tracer_nouv,grid_ens,'gsibundle_fv3lam_ens_tracer_nouv', istatus,&
+                names2d=fv3lam_ens_io_tracermetvars2d_nouv,names3d=fv3lam_ens_io_tracermetvars3d_nouv)
+    else
+       call gsi_bundlecreate(gsibundle_fv3lam_ens_tracer_nouv,grid_ens,'gsibundle_fv3lam_ens_tracer_nouv',istatus, &
+                names3d=fv3lam_ens_io_tracermetvars3d_nouv)
+    endif
+     
+  
 
 
+     
     if(fv3sar_ensemble_opt == 0 ) then  
-      call gsi_fv3ncdf_readuv(dynvars,g_u,g_v)
+      call gsi_fv3ncdf_readuv(grd_fv3lam_ens_uv,g_u,g_v,fv3_filenameginput)
     else
-      call gsi_fv3ncdf_readuv_v1(dynvars,g_u,g_v)
+      call gsi_fv3ncdf_readuv_v1(grd_fv3lam_ens_uv,g_u,g_v,fv3_filenameginput)
     endif
     if(fv3sar_ensemble_opt == 0) then
-      call gsi_fv3ncdf_read(dynvars,'T','t',g_tsen,mype_t)
+      call gsi_fv3ncdf_read(grd_fv3lam_ens_dynvar_io_nouv,gsibundle_fv3lam_ens_dynvar_nouv,fv3_filenameginput%dynvars,fv3_filenameginput)
+      call gsi_fv3ncdf_read(grd_fv3lam_ens_tracer_io_nouv,gsibundle_fv3lam_ens_tracer_nouv,fv3_filenameginput%tracers,fv3_filenameginput)
     else
-      call gsi_fv3ncdf_read_v1(dynvars,'t','T',g_tsen,mype_t)
+      call gsi_fv3ncdf_read_v1(grd_fv3lam_ens_dynvar_io_nouv,gsibundle_fv3lam_ens_dynvar_nouv,fv3_filenameginput%dynvars,fv3_filenameginput)
+      call gsi_fv3ncdf_read_v1(grd_fv3lam_ens_tracer_io_nouv,gsibundle_fv3lam_ens_tracer_nouv,fv3_filenameginput%tracers,fv3_filenameginput)
```

Comment:
These lines exceed the 132 character limit. Please reduce the number of characters.